src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
```

Comment:
nope, this CCE you're seeing is at the call site: this line ` System.out.println(f.get());` has a magic cast inserted - see the bytecode for get:
```
   ALOAD 0
   GETFIELD org/broadinstitute/hellbender/utils/recalibration/Foo2.o : Ljava/lang/Object;
   ARETURN
```

and for relevant line in main 
```
  INVOKEVIRTUAL org/broadinstitute/hellbender/utils/recalibration/Foo2.get ()Ljava/lang/Object;
  CHECKCAST java/lang/String
```